test/e2e/kubelet.go

```diff
@@ -87,25 +90,81 @@ func waitTillNPodsRunningOnNodes(c *client.Client, nodeNames sets.String, podNam
 	})
 }
 
+func updateNodes(c *client.Client, nodeNames sets.String, labels map[string]string, add bool) {
```

Comment:
updateNodeLabels(c, nodeNames, toAdd map[string]string, toRemove map[string]string) and add a comment that it overwrites labels when adding, and silently ignores not existing ones when removing.